### ISSUE TITLE
Vue 2 project: Getting an error related to the vue-composition-api and vue-demi

### ISSUE THREAD

**Issue by nachodd** [fluent-vue/fluent-vue#775]
I have a Vue 2 application that is using [single-spa](https://single-spa.js.org/). For this app I'm using Vue as an external library, by marking it as externals on the `vue.config.js` file (`configureWebpack.externals: ['vue']`). As recommended by single-spa, for dealing with the Vue library resource I'm using `System.js` and specifying a `systemjs-importmap`, like: 
```
<script type="systemjs-importmap">
      {
        "imports": {
          "vue": "https://cdn.jsdelivr.net/npm/vue@2.6.14/dist/vue.js",
           // other stuff...
        }
     }
</script>
```

Everything seems to be working fine on my project. But I'm getting this error on the console when trying to use the `$t` function on the templates: 

```
[Vue warn]: provide is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup().

warn | @ | VM256402 vue.js:634
  | warn | @ | vue-composition-api.mjs?9241:423
  | getCurrentInstanceForFn | @ | vue-composition-api.mjs?9241:452
  | provide | @ | vue-composition-api.mjs?9241:1551
  | setup | @ | index.mjs?bbef:177
  | mergedSetupFn | @ | vue-composition-api.mjs?9241:2160
  | mergedSetupFn | @ | vue-composition-api.mjs?9241:2160
  | mergedSetupFn | @ | vue-composition-api.mjs?9241:2160
  | mergedSetupFn | @ | vue-composition-api.mjs?9241:2160
  | mergedSetupFn | @ | vue-composition-api.mjs?9241:2160
  | eval | @ | vue-composition-api.mjs?9241:1972
  | activateCurrentInstance | @ | vue-composition-api.mjs?9241:1891
  | initSetup | @ | vue-composition-api.mjs?9241:1970
  | wrappedData | @ | vue-composition-api.mjs?9241:1953
  | getData | @ | vue.js:4758
  | initData | @ | vue.js:4715
  | initState | @ | vue.js:4654
  | Vue._init | @ | vue.js:5014
  | Vue | @ | vue.js:5092
  | eval | @ | single-spa-vue.js?0461:1

 // continues...
```
The fact is that is working. But I have tons of console log messages. I think I found a solution, but I'm not completely sure why this is happening. 
So, I tried to debug the error and it seems that `getCurrentInstanceForFn` of the vue-composition-api is not finding the current instance. As stated on the stack trace, I've noted that this has to do with [this portion of code](https://github.com/fluent-vue/fluent-vue/blob/c5ee255d29549f7aaa4735c4475af57ea16a6d54/src/index.ts#L78): 
```
vue2.mixin({
  setup() {
    provide(RootContextSymbol, rootContext)
  },
})
```

I've noted that the `provide()` function is imported from `vue-demi`. In order to fix this issue,  if instead of defining the `provide` in this way (by using `setup()` and the composition-api), we define it in the old-fashion way: 
```
vue2.mixin({
  provide() {
    return {
      RootContextSymbol: rootContext
    }
  }
});
```
The error on the console disappears. 
From my understanding this bug has to do with how do I load the Vue dependency and because of that is not being recognized by the composition api. Fun fact, I'm also using Pinia and other libraries that rely on the vue-composition-api and this is not happening.

Is there a good reason why we should be using this `provide()` method by vue-demi instead of using the code that I'm suggesting? From my understanding it will be the same as the code that I'm suggesting if only for Vue 2. If you are OK with this, I can submit a PR. 

**To Reproduce**
I've already described the scenario. The key is to have Vue as an external library (on the vue.config.js) and to load it with systemjs-importmap. I'm wasn't able to correctly reproduce it on a codesandbox environment, but I can create a test repository if it worth the case.

**Comment by Demivan** [comment]
Thanks for such an extensive investigation.

> Is there a good reason why we should be using this `provide()` method by vue-demi instead of using the code that I'm suggesting?

I have tried using Vue 2 style provide - it does not work with vue-demi. It cannot find the injection inside `i18n` component.

```
[Vue warn]: Injection "Symbol(root-context)" not found.
```

Provided stack trace is indeed strange. You can actually see composition-api plugin setting current instance using `activateCurrentInstance`.

Is your project public? So I can investigate this.

**Comment by Demivan** [comment]
My bad. This actually works:

```js
vue2.mixin({
  provide () {
    return {
      [RootContextSymbol as symbol]: rootContext
    }
  }
})
```

PR is welcome :)

**Comment by nachodd** [comment]
I'm afraid to say that the project I'm working on is not public. I will submit the PR in a while. Thank you very much for the quick response!!!

**Comment by nachodd** [comment]
PR Sent! 🚀

**Comment by Demivan** [comment]
Thank you for contribution. Released it in v3.0.2

You use-case is not very common so there could be more problems. Feel free to open new issues :)
